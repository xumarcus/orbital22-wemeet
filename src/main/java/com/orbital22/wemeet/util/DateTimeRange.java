package com.orbital22.wemeet.util;

import lombok.NonNull;

import java.time.LocalDateTime;

public interface DateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  default boolean contains(DateTimeRange other) {
    return getStartDateTime().isBefore(other.getStartDateTime())
        && getEndDateTime().isAfter(other.getEndDateTime());
  }
}
