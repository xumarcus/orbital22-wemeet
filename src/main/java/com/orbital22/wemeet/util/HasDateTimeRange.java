package com.orbital22.wemeet.util;

import lombok.NonNull;

import java.time.LocalDateTime;

public interface HasDateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  static boolean isValid(HasDateTimeRange dateTimeRange) {
    return dateTimeRange.getStartDateTime().isBefore(dateTimeRange.getEndDateTime());
  }

  default boolean containsRange(HasDateTimeRange other) {
    return getStartDateTime().isBefore(other.getStartDateTime())
        && getEndDateTime().isAfter(other.getEndDateTime());
  }
}
