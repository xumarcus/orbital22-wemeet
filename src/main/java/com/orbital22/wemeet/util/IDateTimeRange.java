package com.orbital22.wemeet.util;

import lombok.NonNull;
import org.apache.commons.lang3.Range;

import java.time.LocalDateTime;
import java.time.chrono.ChronoLocalDateTime;

import static org.apache.commons.lang3.Range.between;

public interface IDateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  default boolean isValid() {
    return getStartDateTime().isBefore(getEndDateTime());
  }

  default Range<ChronoLocalDateTime<?>> getChronoLocalDateTimeRange() {
    return between(getStartDateTime(), getEndDateTime());
  }
}
