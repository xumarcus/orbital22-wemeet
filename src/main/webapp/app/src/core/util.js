// In development, resp.type === 'cors'
// In production, resp.type is per normal
// TODO handle fullyAuthenticated

const ajax = (method, data) => async (uri) => {
    // Spring POST /login accepts formData by default.
    // POST /logout does not take any parameters.
    const isFormData = uri.endsWith("login");
    const resp = await fetch(uri, {
        body: isFormData ? data : JSON.stringify(data),
        headers: {
            "X-XSRF-TOKEN": csrfToken(),
            ...(isFormData ? undefined : {"Content-Type": "application/json"})
        },
        method,
        redirect: "follow",
    });
    if (resp.ok) {
        const location = resp.headers.get('Location');
        if (location != null) {
            // Fetch created resource
            return ajax('GET')(location);
        }
        return resp.json();
    } else {
        throw new Error(resp.statusText, { cause: await resp.json() });
    }
};

const csrfToken = () => document.cookie.replace(/(?:^|.*;\s*)XSRF-TOKEN\s*=\s*([^;]*).*$|^.*$/, '$1');

export default ajax;