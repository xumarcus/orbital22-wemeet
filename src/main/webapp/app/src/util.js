const ajax = (method) => (uri) => {
    const headers = {'Content-Type': 'application/json'};
    return fetch(uri, { method, headers })
        .then(resp => resp.json());
}
export default ajax;