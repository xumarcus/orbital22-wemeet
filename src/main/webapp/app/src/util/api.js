import { useEffect } from 'react';

export function ajax(method, uri) {
    return fetch(uri, { method, headers: {'Content-Type': 'application/json'} });
}

export function useAjax(method, uri, setState, callback) {
    useEffect(() => {
        const fetchData = async () => {
            const response = await ajax(method, uri);
            if (response.ok) {
                const json = await response.json();
                setState(json);
            } else {
                callback();
            }
        }
        fetchData().then();
    }, []);
}