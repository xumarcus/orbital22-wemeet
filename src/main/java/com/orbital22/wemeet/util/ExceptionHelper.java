package com.orbital22.wemeet.util;

import org.jetbrains.annotations.Nullable;
import org.springframework.data.rest.core.RepositoryConstraintViolationException;
import org.springframework.validation.Errors;
import org.springframework.validation.MapBindingResult;

import java.util.HashMap;

public interface ExceptionHelper {
  static RepositoryConstraintViolationException from(
          Object object, @Nullable String field, String errorCode) {
    return from(object, field, errorCode, null);
  }

  static RepositoryConstraintViolationException from(
          Object object, @Nullable String field, String errorCode, @Nullable String defaultMessage) {
    Errors errors = new MapBindingResult(new HashMap<>(), object.getClass().getName());
    Object[] errorArgs = new Object[] {object};

    if (field != null) {
      errors.rejectValue(field, errorCode, errorArgs, defaultMessage);
    } else {
      errors.reject(errorCode, errorArgs, defaultMessage);
    }

    return new RepositoryConstraintViolationException(errors);
  }
}
