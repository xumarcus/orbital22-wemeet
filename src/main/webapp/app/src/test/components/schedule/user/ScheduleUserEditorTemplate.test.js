import React from 'react'
import { render } from '@testing-library/react'

import {
  getSelfRank,
  ScheduleUserEditorTemplateInner
} from '../../../../components/schedule/user/ScheduleUserEditorTemplate'
import { SAMPLE_EVENT } from '../../../mocks/events'
import { setupForSyncfusionTest } from '../../../test-core/util'

describe('ScheduleUserEditorTemplate', () => {
  beforeAll(setupForSyncfusionTest)

  it('should render correctly without rank', () => {
    const { asFragment } = render(<ScheduleUserEditorTemplateInner
      id={1}
      rank={getSelfRank({ id: 1 }, [])}
      {...SAMPLE_EVENT}
                                  />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly with info from self', () => {
    const { asFragment, queryByTestId } = render(<ScheduleUserEditorTemplateInner
      id={1}
      rank={getSelfRank({ id: 1 }, [{
        user: { id: 1 },
        rank: 1989
      }])}
      available
      {...SAMPLE_EVENT}
                                                 />)
    expect(queryByTestId('rank')?.getAttribute('value')).toEqual('1989')
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly with info not from self', () => {
    const { asFragment, queryByTestId } = render(<ScheduleUserEditorTemplateInner
      id={1}
      rank={getSelfRank({ id: 1 }, [{
        user: { id: 2 },
        rank: 8964
      }])}
      available
      {...SAMPLE_EVENT}
                                                 />)
    expect(queryByTestId('rank')?.getAttribute('value')).toBeFalsy()
    expect(asFragment()).toMatchSnapshot()
  })
})
