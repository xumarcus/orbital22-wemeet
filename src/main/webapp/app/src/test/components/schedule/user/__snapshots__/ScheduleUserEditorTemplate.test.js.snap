// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleUserEditorTemplate should render correctly with info from self 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="timeSlotId"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="capacity"
            disabled=""
            id="capacity"
            name="capacity"
            style="width: 100%;"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Available
        </td>
        <td
          colspan="4"
        >
          <div
            aria-checked="true"
            class="e-switch-wrapper e-wrapper"
          >
            <input
              class="e-field e-control e-switch e-lib"
              data-name="available"
              id="available"
              type="checkbox"
            />
            <span
              class="e-switch-inner e-switch-active"
            >
              <span
                class="e-switch-on"
              />
              <span
                class="e-switch-off"
              />
            </span>
            <span
              class="e-switch-handle e-switch-active"
            />
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Rank
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="rank"
            id="rank"
            name="rank"
            style="width: 100%;"
            type="number"
            value="1989"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="startDateTime"
              disabled="disabled"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="endDateTime"
              disabled="disabled"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ScheduleUserEditorTemplate should render correctly with info not from self 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="timeSlotId"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="capacity"
            disabled=""
            id="capacity"
            name="capacity"
            style="width: 100%;"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Available
        </td>
        <td
          colspan="4"
        >
          <div
            aria-checked="true"
            class="e-switch-wrapper e-wrapper"
          >
            <input
              class="e-field e-control e-switch e-lib"
              data-name="available"
              id="available"
              type="checkbox"
            />
            <span
              class="e-switch-inner e-switch-active"
            >
              <span
                class="e-switch-on"
              />
              <span
                class="e-switch-off"
              />
            </span>
            <span
              class="e-switch-handle e-switch-active"
            />
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Rank
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="rank"
            id="rank"
            name="rank"
            style="width: 100%;"
            type="number"
            value=""
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="startDateTime"
              disabled="disabled"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="endDateTime"
              disabled="disabled"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ScheduleUserEditorTemplate should render correctly without rank 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="timeSlotId"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="capacity"
            disabled=""
            id="capacity"
            name="capacity"
            style="width: 100%;"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Available
        </td>
        <td
          colspan="4"
        >
          <div
            aria-checked="false"
            class="e-switch-wrapper e-wrapper"
          >
            <input
              class="e-field e-control e-switch e-lib"
              data-name="available"
              id="available"
              type="checkbox"
            />
            <span
              class="e-switch-inner"
            >
              <span
                class="e-switch-on"
              />
              <span
                class="e-switch-off"
              />
            </span>
            <span
              class="e-switch-handle"
            />
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Rank
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            data-testid="rank"
            disabled=""
            id="rank"
            name="rank"
            style="width: 100%;"
            type="number"
            value=""
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="startDateTime"
              disabled="disabled"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="endDateTime"
              disabled="disabled"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
